AudioLibrary.WMME/WaveOut.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using AudioLibrary.WMME.Native;
+
+namespace AudioLibrary.WMME
+{
+    internal class WaveOut
+    {
+        private IntPtr m_WaveOut;
+        private WaveOutBuffer m_Buffers; // linked list
+        private WaveOutBuffer m_CurrentBuffer;
+        private Thread m_Thread;
+        private BufferFillEventHandler m_FillProc;
+		private bool m_Finished;
+		private byte m_zero;
+
+        private WaveNative.WaveDelegate m_BufferProc = new WaveNative.WaveDelegate(WaveOutBuffer.WaveOutProc);
+
+        public WaveOut(int device, WAVEFORMATEX format, int bufferSize, int bufferCount, BufferFillEventHandler fillProc)
+        {
+			m_zero = format.wBitsPerSample == 8 ? (byte)128 : (byte)0;
+            m_FillProc = fillProc;
+
+            var errorCode = (MMErrors)WaveNative.waveOutOpen(out m_WaveOut, device, ref format, m_BufferProc, 0, (int)CallBackFlag.CALLBACK_FUNCTION);
+            if (errorCode != MMErrors.MMSYSERR_NOERROR)
+                throw new MixerException(errorCode, Audio.GetErrorDescription(FuncName.fnWaveOutOpen, errorCode));
+
+            AllocateBuffers(bufferSize, bufferCount);
+            m_Thread = new Thread(new ThreadStart(ThreadProc));
+            m_Thread.Start();
+        }
+
+        ~WaveOut()
+        {
+            Dispose();
+        }
+
+        public void Dispose()
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Either implement 'IDisposable.Dispose', or totally rename this method to prevent confusion. ([csharpsquid:S2953](https://sonarcloud.io/organizations/alex-yevtushenko3820/rules#rule_key=csharpsquid%3AS2953))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=IPP&pullRequest=7&issues=AWfG1-gp8DroMTOoScWl&open=AWfG1-gp8DroMTOoScWl)

[ISSUE_KEY]: AWfG1-gp8DroMTOoScWl